### PR TITLE
Add new CAPA e2e conformance jobs

### PR DESCRIPTION
**Pull request by detiber**
Related to: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1360

/assign @ncdc

### CODE REVIEW

**Comment by detiber** [comment]
bump

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml [hunk 1]
```diff
@@ -83,6 +83,38 @@ periodics:
...
+    - org: kubernetes-sigs
+      repo: cluster-api-provider-aws
+      base_ref: release-0.4
+      path_alias: "sigs.k8s.io/cluster-api-provider-aws"
+  spec:
+    containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20191112-9f04410-1.16
+        command:
+          - "runner.sh"
+          - "./scripts/ci-e2e-conformance.sh"
```
Comment (ncdc) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: Should this be ci-conformance.sh?

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml [hunk 2]
```diff
@@ -304,6 +336,36 @@ postsubmits:
...
+      preset-aws-ssh: "true"
+      preset-aws-credential: "true"
+    branches:
+      - release-0.4
+    spec:
+      containers:
+        - image: gcr.io/k8s-testimages/kubekins-e2e:v20191112-9f04410-1.16
+          command:
+            - "runner.sh"
+            - "./scripts/ci-e2e-conformance.sh"
```
Comment (ncdc) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: Should this be ci-conformance.sh?

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-presubmits.yaml [hunk 3]
```diff
@@ -152,3 +152,32 @@ presubmits:
...
+      preset-kind-volume-mounts: "true"
+      preset-service-account: "true"
+      preset-aws-ssh: "true"
+      preset-aws-credential: "true"
+    spec:
+      containers:
+        - image: gcr.io/k8s-testimages/kubekins-e2e:v20191112-9f04410-1.16
+          command:
+            - "runner.sh"
+            - "./scripts/ci-e2e-conformance.sh"
```
Comment (ncdc) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-presubmits.yaml]: ci-conformance.sh?

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml (hunk 2 above)
Comment (detiber) [reply config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: No, see https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1360/files#diff-a2c024158251d83fd27abc4413253616R1

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-presubmits.yaml (hunk 3 above)
Comment (detiber) [reply config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-presubmits.yaml]: No, see https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1360/files#diff-a2c024158251d83fd27abc4413253616R1

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml (hunk 1 above)
Comment (detiber) [reply config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: No, see https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1360/files#diff-a2c024158251d83fd27abc4413253616R1

**Comment by ncdc** [comment]
Ok, so we need to forward-port https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1360 to master?

**Comment by detiber** [comment]
> Ok, so we need to forward-port kubernetes-sigs/cluster-api-provider-aws#1360 to master?

Eventually, yes. The current periodic and postsubmit jobs only target the `release-0.4` branch.

**Comment by ncdc** [comment]
Thanks for being patient with me as I multi-task poorly 😄 
/lgtm

**Pull request merged**